zebra-chain/src/transaction/hash.rs

```diff
@@ -51,7 +66,19 @@ mod tests {
 
         assert_eq!(
             format!("{:?}", hash),
-            "TransactionHash(\"bf46b4b5030752fedac6f884976162bbfb29a9398f104a280b3e34d51b416631\")"
+            r#"TransactionHash("bf46b4b5030752fedac6f884976162bbfb29a9398f104a280b3e34d51b416631")"#
+        );
+    }
+
+    #[test]
+    fn transactionhash_from_str() {
+        let hash = TransactionHash::from_str(
+            "bf46b4b5030752fedac6f884976162bbfb29a9398f104a280b3e34d51b416631",
+        );
+        assert!(hash.is_ok());
+        assert_eq!(
+            format!("{:?}", hash.unwrap()),
```

Comment:
If you apply the above, you have already unwrapped.

```suggestion
            format!("{:?}", hash),
```